src/main/java/com/treasure/hunt/strategy/searcher/impl/strategyFromPaper/RoutinesFromPaper.java

```diff
@@ -0,0 +1,336 @@
+package com.treasure.hunt.strategy.searcher.impl.strategyFromPaper;
+
+import com.treasure.hunt.strategy.hint.impl.HalfPlaneHint;
+import com.treasure.hunt.strategy.searcher.Movement;
+import com.treasure.hunt.utils.JTSUtils;
+import org.locationtech.jts.geom.Coordinate;
+import org.locationtech.jts.geom.LineSegment;
+import org.locationtech.jts.geom.Point;
+import org.locationtech.jts.geom.util.AffineTransformation;
+
+import java.util.Arrays;
+
+import static com.treasure.hunt.strategy.hint.impl.HalfPlaneHint.Direction.right;
+import static com.treasure.hunt.strategy.hint.impl.HalfPlaneHint.Direction.up;
+import static com.treasure.hunt.strategy.searcher.impl.strategyFromPaper.GeometricUtils.*;
+import static com.treasure.hunt.utils.JTSUtils.lineWayIntersection;
+
+/**
+ * TODO explain what rho, sigma and phi do in the paper
+ *
+ * @author bsen
+ */
+public class RoutinesFromPaper {
+    static Movement rectangleScan(Point A, Point B, Point C, Point D, Movement move) {
+        return rectangleScan(A.getCoordinate(), B.getCoordinate(), C.getCoordinate(), D.getCoordinate(), move);
+    }
+
+    static Movement rectangleScan(Coordinate A, Coordinate B, Coordinate C, Coordinate D, Movement move) {
+        int k = (int) A.distance(B);
+        Point[] a = new Point[k + 1];
+        Point[] b = new Point[k + 1];
+
+        { //create a_i on line segment AB
+            double xDist = B.getX() - A.getX();
+            double yDist = B.getY() - A.getY();
+            for (int i = 0; i <= k; i++) {
+                a[i] = JTSUtils.createPoint(A.getX() + xDist * ((double) i / (double) k), A.getY() + yDist *
+                        ((double) i / (double) k));
+            }
+        }
+        { //create b_i on line segment DC
+            double xDist = C.getX() - D.getX();
+            double yDist = C.getY() - D.getY();
+            //for (int i = 0; i <= k; i++) {
+            for (int i = 0; i <= k; i++) {
+                b[i] = JTSUtils.createPoint(D.getX() + xDist * ((double) i / (double) k), D.getY() + yDist *
+                        ((double) i / (double) k));
+            }
+        }
+
+        if (k % 2 == 1) //code like in paper
+        {
```

Comment:
Tick the format boxes in the commit message or str+alt+l { should be same lie as loop